### PR TITLE
 java interop - triple protocol-java -go

### PR DESCRIPTION
**Pull request by fansofml**
#699

### CODE REVIEW

File: java_interop/protobuf-triple/go/go-client/cmd/client.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// export DUBBO_GO_CONFIG_PATH=$PATH_TO_SAMPLES/rpc/triple/pb/dubbogo-java/go-client/conf/dubbogo.yml
+func main() {
+
+	cli, err := client.NewClient(
+		client.WithClientURL("127.0.0.1:36969"),
+	)
+	if err != nil {
+		panic(err)
+	}
+	if err != nil {
```
Comment (FoghostCn) [inline java_interop/protobuf-triple/go/go-client/cmd/client.go:37]: plz remove no used code

Comment (fansofml) [reply java_interop/protobuf-triple/go/go-client/cmd/client.go:37]: 好的，我去修改一下

File: java_interop/protobuf-triple/java/java-client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                <configuration>
+                    <!--suppress UnresolvedMavenProperty -->
+                    <protocArtifact>com.google.protobuf:protoc:3.19.4:exe:${os.detected.classifier}</protocArtifact>
+                    <protocPlugins>
+                        <protocPlugin>
+                            <id>dubbo</id>
+                            <groupId>org.apache.dubbo</groupId>
+                            <artifactId>dubbo-compiler</artifactId>
+                            <version>${dubbo.version}</version>
+                            <mainClass>org.apache.dubbo.gen.tri.Dubbo3TripleGenerator</mainClass>
```
Comment (FoghostCn) [inline java_interop/protobuf-triple/java/java-client/pom.xml:64]: plz remove the generate code and add doc how to generated them

Comment (fansofml) [reply java_interop/protobuf-triple/java/java-client/pom.xml:65]: I have added instructions on how to generate proto in the README.

File: java_interop/protobuf-triple/java/java-server/run.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+mvn -e clean compile exec:java -Dexec.mainClass="org.apache.dubbo.sample.Provider"
```
Comment (chickenlj) [inline java_interop/protobuf-triple/java/java-server/run.sh:1]: Better use `mvn wrapper`

File: java_interop/protobuf-triple/java/java-server/src/main/java/org/apache/dubbo/sample/Provider.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+public class Producer {
+    public static void main(String[] args) throws IOException {
+        ServiceConfig<Greeter> service =  new ServiceConfig<>();
+        service.setInterface(Greeter.class);
+        service.setRef(new GreeterImpl());
+
+        DubboBootstrap bootstrap = DubboBootstrap.getInstance();
+        bootstrap.application(new ApplicationConfig("java-go-sample-server"))
+                .registry(new RegistryConfig("zookeeper://127.0.0.1:2181"))
```
Comment (chickenlj) [inline java_interop/protobuf-triple/java/java-server/src/main/java/org/apache/dubbo/sample/Provider.java:20]: registry is not mandatory for this case

File: java_interop/protobuf-triple/java/java-client/src/main/java/org/apache/dubbo/sample/Consumer.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    public static void main(String[] args) {
+        DubboBootstrap bootstrap = DubboBootstrap.getInstance();
+        ReferenceConfig<Greeter> ref = new ReferenceConfig<>();
+        ref.setInterface(Greeter.class);
+        ref.setProtocol(CommonConstants.TRIPLE);
+        ref.setProxy(CommonConstants.NATIVE_STUB);
+        ref.setTimeout(3000);
+        bootstrap.application(new ApplicationConfig("duboo-test"))
+                .registry(new RegistryConfig("zookeeper://127.0.0.1:2181"))
+                .protocol(new ProtocolConfig(CommonConstants.TRIPLE,50052))
```
Comment (chickenlj) [inline java_interop/protobuf-triple/java/java-client/src/main/java/org/apache/dubbo/sample/Consumer.java:20]: protocol is not needed for consumer, remove this line please

File: java_interop/protobuf-triple/java/java-client/src/main/java/org/apache/dubbo/sample/Consumer.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+public class ApiConsumer {
+    public static void main(String[] args) {
+        DubboBootstrap bootstrap = DubboBootstrap.getInstance();
+        ReferenceConfig<Greeter> ref = new ReferenceConfig<>();
+        ref.setInterface(Greeter.class);
+        ref.setProtocol(CommonConstants.TRIPLE);
+        ref.setProxy(CommonConstants.NATIVE_STUB);
+        ref.setTimeout(3000);
+        bootstrap.application(new ApplicationConfig("duboo-test"))
+                .registry(new RegistryConfig("zookeeper://127.0.0.1:2181"))
```
Comment (chickenlj) [inline java_interop/protobuf-triple/java/java-client/src/main/java/org/apache/dubbo/sample/Consumer.java:19]: registry is not necessary for this case, better use `ref.setUrl("tri://ip:port")` instead.

File: java_interop/protobuf-triple/go/go-server/cmd/server.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+	resp := &greet.HelloReply{Message: req.Name}
+	return resp, nil
+}
+
+func main() {
+	ins, err := dubbo.NewInstance(
+		dubbo.WithName("org.apache.dubbo.sample.GreeterImpl"),
+		dubbo.WithRegistry(
+			registry.WithZookeeper(),
+			registry.WithAddress("127.0.0.1:2181"),
```
Comment (chickenlj) [inline java_interop/protobuf-triple/go/go-server/cmd/server.go:44]: remove this line

File: java_interop/protobuf-triple/go/go-client/conf/dubbogo.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+dubbo:
+  registries:
+    zk:
+      address: zookeeper://127.0.0.1:2181
+  Consumer:
+    services:
+      GreeterConsumer:
+        # interface is for registry
+        interface: org.apache.dubbo.demo.GreeterImpl
```
Comment (chickenlj) [inline java_interop/protobuf-triple/go/go-client/conf/dubbogo.yml:9]: Is this file still needed? I think we don't need this file anymore with the new API. Please confirm if we can delete this file.

File: java_interop/protobuf-triple/go/go-server/conf/dubbogo.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+      address: zookeeper://127.0.0.1:2181
+  protocols:
+    triple:
+      name: tri
+      port: 50052
+  provider:
+    services:
+      GreeterProvider:
+        # interface is for registry
+        interface: org.apache.dubbo.demo.GreeterImpl
```
Comment (chickenlj) [inline java_interop/protobuf-triple/go/go-server/conf/dubbogo.yml:13]: Please confirm if we can delete this file.

File: java_interop/protobuf-triple/java/java-client/pom.xml [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo</artifactId>
+            <version>${dubbo.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-dependencies-zookeeper-curator5</artifactId>
+            <type>pom</type>
+            <version>${dubbo.version}</version>
+        </dependency>
```
Comment (chickenlj) [inline java_interop/protobuf-triple/java/java-client/pom.xml:34]: remove this dependency after `registry` configuration removed.

File: java_interop/protobuf-triple/java/pom.xml [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo</artifactId>
+            <version>${dubbo.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-dependencies-zookeeper-curator5</artifactId>
+            <type>pom</type>
+            <version>${dubbo.version}</version>
+        </dependency>
```
Comment (chickenlj) [inline java_interop/protobuf-triple/java/pom.xml:34]: Remove this dependency

File: java_interop/protobuf-triple/java/java-server/run.sh (hunk 3 above)
Comment (chickenlj) [reply java_interop/protobuf-triple/java/java-server/run.sh:1]: https://github.com/apache/dubbo-samples/tree/master/11-quickstart

check mvn wrapper configuration in this directory.

**Review by FoghostCn — APPROVED** [review]

**Comment by EndlessSeeker** [comment]
plz add integrate test for this case

**Comment by fansofml** [comment]
> plz add integrate test for this case

You can test it by following the instructions in the README.
